### PR TITLE
[VisionEncoderDecoderModel] Update loss function

### PR DESCRIPTION
**Pull request by NielsRogge**
# What does this PR do?

Models like Donut are currently broken on main, they can't be fine-tuned. In order to unblock users at #39473, this PR reverts #36753.

It looks like the `ForCausalLMLoss` class shifts the labels, however the VisionEncoderDecoderModel class does not expect shifted labels as seen [here](https://github.com/huggingface/transformers/blob/d42e96a2a731c4a772e396baa0d915524c873ff0/src/transformers/models/vision_encoder_decoder/modeling_vision_encoder_decoder.py#L547).

### CODE REVIEW

**Comment by NielsRogge** [comment]
cc @ArthurZucker this one can be merged

**Review by SunMarc — APPROVED** [review]
Thanks
